### DISCUSSION TITLE
BOUNTY: Get all subfolders from msn.com

### DISCUSSION THREAD

**Discussion by rcmaehl** [rcmaehl/MSEdgeRedirect#338] [Help Bounties]
News redirection currently needs serious improvement. To do so, I need to figure out all the news subfolders on msn.com ( e.g. msn.com/techandscience or msn.com/autos/enthusiasts ).

Looking for a simple list of all subfolders from MSN.com 

BOUNTY; $50 USD PAYPAL

**Comment by whamer100** [comment]
There seems to be an api endpoint for resolving data that gets displayed when viewing msn.com. Viewing `https://www.msn.com/resolver/api/resolve/v3/config/?expType=AppConfig&expInstance=default&apptype=hub&v=20231109.593` responds with a json object with a plethora of data. At `configs.shared/msn-ns/NavigationData/news` you can see the list of tabs, and at `shared/msn-ns/CategoryData/default` you can find a list of all the categories (for example, the categoryDocumentLink `autos` refers to `https://www.msn.com/en-us/news/autos`)

**Reply by rcmaehl** [reply]
I sent you this and the other bounty. It looks like subcategories (or at least some of them (e.g. `msn.com/en-us/entertainment/comics`) are not listed in the json however. I can probably work around that but let me know if you find somewhere else that lists the categories and subcategories.

**Reply by whamer100** [reply]
Thank you! And sure thing, I'll take another look later and see if i can find more categories. MSN's apis are pretty arbitrary.

**Reply by rcmaehl** [reply]
Sounds good. Thank you!

**Comment by Ozaku** [comment]
Hey so fun fact - I saw the thing about this being blacklisted on Google, but their AI still told me about this :)
